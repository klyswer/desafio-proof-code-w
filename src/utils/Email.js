class Email {
  constructor() {
    this.regExp = /^(([^<>()[\]\\.,;:\s@"]+(\.[^<>()[\]\\.,;:\s@"]+)*)|(".+"))@((\[[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}])|(([a-zA-Z\-0-9]+\.)+[a-zA-Z]{2,}))$/;
  }

  isValidEmail = _email => {
    return this.regExp.test(_email);
  };
}

export default new Email();
