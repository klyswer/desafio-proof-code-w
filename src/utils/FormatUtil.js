

/**
*  Este método recibe una cadena numerica, retorna la 
*  cadena con formato de moneda local CLP.
*/
export const agregarformatoPesos = (numero) => {
  const formatter = new Intl.NumberFormat("es-CL", {
    style: "currency",
    currency: "CLP"
  });
  return formatter.format(numero);
}

/**
 * Método que recibe una cadena  de caracteres "Descripción"
 * Devuelve el parrafo cortado según el número.
 */
export const reducirDescripcion = (descripcion) => {
  const separaCaracteres = descripcion.split("");
  const numeroCaracteres = separaCaracteres.length;
  if (numeroCaracteres > 25) {
    const resto = numeroCaracteres-25
    separaCaracteres.splice(25, resto, ".", ".",".");
  return separaCaracteres.toString().replace(/,/g, "");
  }
    return separaCaracteres.toString().replace(/,/g, "");  
};




    // const sumarPorMarca = (marca,listaSeleccionados)=>{
    //   const sacarTotalPorMarca = listaSeleccionados.reduce((acumulador, valorActual) => {
    //     const elementoYaExiste = acumulador.find(elemento => elemento.brand === valorActual.brand);
    //     if (elementoYaExiste) {
    //       return acumulador.map((elm) => {
    //         if (elm.brand === valorActual.brand) {
    //           return {
    //             ...elm,
    //             acumulado: elm.price + valorActual.price
    //           }
    //         }
    //         return elm;
    //       });
    //     } 
    //     return [...acumulador, valorActual];
    //   }, []);
    //   return sacarTotalPorMarca;
    // }
