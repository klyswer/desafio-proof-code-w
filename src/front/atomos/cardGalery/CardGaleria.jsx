import React from 'react';
import PropTypes from 'prop-types';
import {agregarformatoPesos} from '../../../utils/FormatUtil';
import './style.css';
import ItemTipoEntrega from '../itemTipoEntrega/ItemTipoEntrega';

const CardGaleria = ({src,descripcion,precio,id,marca,agregarProducto}) => {
  return (
    <div cat={marca}>
      <div id={id} className="card card_mod h-100">

        <img className="card-img-top tamaÃ±o_imagenes" src={src!==''&& `https://${src}`} alt={descripcion} />

        <div className="line_head">
          <div className="container_card_texto">
            <div className="caja_texto_cards">
              <strong>{`${marca} `}</strong> 
              <span className="descrip_span">{descripcion}</span>
            </div>
          </div>

          <div className="precio_card">
            <span>{agregarformatoPesos(precio)}</span>
          </div>

        </div>          
        <div className="container_tipo_entrega">
          <ItemTipoEntrega />
          <ItemTipoEntrega tipo="retiro" texto="retiro" />
        </div>
        <div className="card-footer card_foot_mod">
          <button type="button" onClick={agregarProducto}>Agregar</button>
        </div>
      </div>
    </div> 
  );
};

CardGaleria.defaultProps = {
  agregarProducto:()=>null
}

CardGaleria.propTypes = {
  src:PropTypes.string.isRequired,
  descripcion:PropTypes.string.isRequired,
  precio:PropTypes.number.isRequired,
  id:PropTypes.string.isRequired,
  marca:PropTypes.string.isRequired,
  agregarProducto: PropTypes.func
};


export default CardGaleria;