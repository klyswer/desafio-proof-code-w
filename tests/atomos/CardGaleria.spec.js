import React from 'react';
/* eslint-disable no-undef */
import { shallow } from 'enzyme';
import CardGaleria from '../../src/front/atomos/cardGalery/CardGaleria';
import { agregarformatoPesos } from '../../src/utils/FormatUtil';


describe('<CardGaleria />', () => {

  const img = 'aire.svg';
  const descript = "Producto Prueba";

  test('Primer SnapShot:<CardGaleria />', () => {
    const wrapper = shallow( <CardGaleria src={img} descripcion={descript} precio={1000} id="01k2021" marca="Marca01" />);
    expect(wrapper).toMatchSnapshot();
  });


  test('Caso prueba 1: props "src", debe renderizar el "string:ruta de src" recibido por props.', () => {
    const wrapper = shallow( <CardGaleria src={img} descripcion={descript} precio={1000} id="01k2021" marca="Marca01" />);
    const imgRender = wrapper.find('img').prop('src');

    expect(imgRender.includes(img)).toBe(true);
  });


  test('Caso prueba 2: props "descripcion", recibe un "string" y renderiza un svg.', () => {
    const texto = 'Descripci√≥n del producto.';
    const wrapper = shallow(<CardGaleria src={img} descripcion={texto} precio={1000} id="01k2021" marca="Marca01" />);
    const descripcionEsperada = wrapper.find('.descrip_span').text();

    expect(descripcionEsperada).toEqual(texto);
  });
  
  test('Caso prueba 3: props "Precio", recibe un "number" y lo muestra como precio.',()=>{
    const precioNumero = 1500;
    const valorFormateado = agregarformatoPesos(precioNumero);
    const wrapper =shallow(<CardGaleria src={img} descripcion={descript} precio={precioNumero} id="01k2021" marca="Marca01" />);
    const precioEsperado = wrapper.find('.precio_card span').text();

    expect(precioEsperado).toEqual(valorFormateado);
  });

  test('Caso prueba 4: props "id", recibe un "string" y asigna la propiedad "id" en el div del "cardGaleria".', () => {
    const ide = 'kloikd03942gf';
    const wrapper = shallow(<CardGaleria src={img} descripcion={descript} precio={1000} id={ide} marca="Marca01" />);
    const idEsperado = wrapper.find('div.card_mod').prop('id');

    expect(idEsperado).toEqual(ide);
  });

  test('Caso prueba 5: props "marca", recibe un "string" y renderiza una marca.', () => {
    const marcaPrueba = 'Marca09';
    const marcaFormateada= `${marcaPrueba} `;
    const wrapper = shallow(<CardGaleria src={img} descripcion={descript} precio={1000} id="01k2021" marca={marcaPrueba} />);
    const marcaEsperada = wrapper.find('.caja_texto_cards strong').text();

    expect(marcaEsperada).toEqual(marcaFormateada);
  });
  
});