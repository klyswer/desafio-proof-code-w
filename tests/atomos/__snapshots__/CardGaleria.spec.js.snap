// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CardGaleria /> Primer SnapShot: <CardGaleria /> 1`] = `
<div
  cat="Marca01"
>
  <div
    className="card card_mod h-100"
    id="01k2021"
  >
    <img
      alt="Producto Prueba"
      className="card-img-top tamaño_imagenes"
      src="https://aire.svg"
    />
    <div
      className="line_head"
    >
      <div
        className="container_card_texto"
      >
        <div
          className="caja_texto_cards"
        >
          <strong>
            Marca01 
          </strong>
          <span
            className="descrip_span"
          >
            Producto Prueba
          </span>
        </div>
      </div>
      <div
        className="precio_card"
      >
        <span>
          CLP 1,000
        </span>
      </div>
    </div>
    <div
      className="container_tipo_entrega"
    >
      <ItemTipoEntrega
        texto="despacho"
        tipo="despacho"
      />
      <ItemTipoEntrega
        texto="retiro"
        tipo="retiro"
      />
    </div>
    <div
      className="card-footer card_foot_mod"
    >
      <button
        onClick={[Function]}
        type="button"
      >
        Agregar
      </button>
    </div>
  </div>
</div>
`;
